libyara/modules/math/math.c

```diff
@@ -715,6 +717,53 @@ define_function(mode_global)
   return_integer(most_common);
 }
 
+define_function(inttostring)
+{
+  int64_t i = integer_argument(1);
+  char *str = NULL;
+  asprintf(&str, "%lld", i);
+  return_string(str == NULL ? (char*) YR_UNDEFINED : str);
+}
+
+define_function(inttostring_base)
+{
+  int64_t i = integer_argument(1);
+  int64_t base = integer_argument(2);
+  char *str = NULL;
+  char *fmt;
+  switch (base)
+  {
+  case 10:
+    fmt = "%lld";
+    break;
+  case 8:
+    fmt = "%llo";
+    break;
+  case 16:
+    fmt = "%llx";
+    break;
+  default:
+    return_string(YR_UNDEFINED);
+  }
+  asprintf(&str, fmt, i);
```

Comment:
address-sanitizer has found a memory leak here, `asprintf` allocates a memory buffer that is never released. `return_string` makes its own copy of the string, that's why in other places where `return_string` is used, it receives a pointer to a local variable in the stack. We would use a large enough local buffer that holds the number in string form and usee `sprintf` instead.

Reply:
Good catch! I misread the `return_string` macro. I am now allocating enough space on the stack and using `snprintf()`.